src/core/net/dns_client.cpp

```diff
@@ -753,6 +753,9 @@ Client::Client(Instance &aInstance)
     static_assert(kServiceQuerySrv == 3, "kServiceQuerySrv value is not correct");
     static_assert(kServiceQueryTxt == 4, "kServiceQuerySrv value is not correct");
 #endif
+#if OPENTHREAD_CONFIG_DNS_CLIENT_OVER_TCP_ENABLE
+    memset(&mSendLink, 0, sizeof(mSendLink));
```

Comment:
Looks good. Suggest using `ClearAllBytes(mSendlink)` instead?
(effectively does the same `memset()` in its implementation, but simpler to read and will determine the proper size automatically)
```suggestion
    ClearAllBytes(mSendLink);
```

Reply:
Updated based on your feedback. Thanks!